### PR TITLE
cli/demo: simplify the TLS configuration

### LINKED ISSUE #92541: Local development and setting stable credentials via cockroach demo
**Is your feature request related to a problem? Please describe.**

CockroachDB is [aiming to remove](https://github.com/cockroachdb/cockroach/issues/53404) the `insecure` flag and suggests `cockroachdb demo` as an automatically secure alternative for developers. What seems to be missing is one reason _why_ app developers turn to the insecure mode now.

Our application developers are developing a web app locally. They need a local cockroachdb node to work against. This is a throw-away database. It runs on their laptop. They might do start and stop one of these databases for each app, multiple times a day. There seem to be two options:

* `cockroach start-single-node --insecure` (deprecated, annoying to use an extra `insecure` flag in every `cockroach sql` invocation)
* `cockroach demo` (random password)

The way `demo` randomizes the password makes local development harder. Developers need to setup the development database, to load in development data, and configure their web app database url to point at the local cockroachdb node. They may be moving between multiple projects, stopping and starting short-lived, single-node dev databases often. App developers tend to put this all behind an automation step ( `up` script, `npm run db`, `make db`, whatever).

**Describe the solution you'd like**

Today, app developers are using insecure mode because they need a predictable, no-fuss local dev database. `cockroach demo` seems like it could do things in a more secure way by default, but having a randomized password is difficult for local scripting.

```
cockroach demo --password password --listen-addr 127.0.0.1:26257 --http-addr 127.0.0.1:5050
```

The connection string is known and can be used in local development setups.

```
postgresql://root:password@127.0.0.1:26257/mydb
```

**Describe alternatives you've considered**

You can use `cockroach demo` today with its randomized password, long enough to create a new user with a known password. But this is an extra step in local development, especially starting and stopping databases for different projects. We're not really guarding against anything here.

**Additional context**

Again, the use case is _local_ development. I'm all for proper TLS certificate setups in real setups. But we just want a mode for developers on their laptops to quickly work on their web apps with a fake throw-away database. 

Similar to: https://github.com/cockroachdb/cockroach/issues/68437
Jira issue: CRDB-21848
Epic: CRDB-22814

### LINKED ISSUE #94309: sql/demo: stabilize the SQL connection URL
Currently every time `cockroach demo` starts the SQL connection URL changes, because the directory where the TLS server cert is stored changed.

It should become the same.

Epic: CRDB-22814

Jira issue: CRDB-22820

### LINKED ISSUE #94310: sql/demo: server TLS cert for system tenant and secondary tenant is different
Also, the URL for a secondary tenant currently uses a separate cert file. This is confusing, and prevents the user from quickly switching between tenants by just using the `cluster` connection option.

Instead, the server cert for secondary tenants should be the same as for the system tenant.

Epic: CRDB-22814

Jira issue: CRDB-22821

### PR DESCRIPTION
**Pull request by knz**
Fixes #94310. 
Fixes #94309. 
Fixes #92541.
Epic: CRDB-22814

TLDR: this patch ensures a separate SQL client can resume the
connection to `cockroach demo` when it is restarted, without
reconfiguring the connection URL. It also enables running `cockroach
sql`against a`demo` session with fewer/simpler arguments.

NB: This does not fix _all_ our woes from CRDB-22814; there will be other PRs to address the remaining issues.

**Change 1 (enhancement): reuse directory**

Prior to this patch, `cockroach demo` was auto-generating new TLS
keys/certs on every run, and using a different directory every time.

Especially the different directory made the tool difficult to use: it
was forcing the connection URLs to change on every invocation.

To fix this, this patch now reuses the same directory,
`$HOME/.cockroach-demo`, for every invocation. Certificate/key files
in that directory are reused if already present.

This also makes it possible for SQL clients to simply reconnect
(reusing the same connection string) when the `demo` session is
restarted.

NB: The reason why this change does not use `$HOME/.cockroach-certs`
is that it should remain possible to run `cockroach demo` alongside a
regular server, and the node certs for a regular server should contain
different metadata than for a demo server (esp. relating to server
addresses).

**Change 2 (enhancement): also produce a TLS client cert for the `demo` user**

Prior to this patch, `demo` would only write a TLS client cert for the
`root` user in the certificate directory. This was preventing direct
use of TLS client cert authentication for the `demo` user,
even by those clients that support TLS client cert authn (such as
`cockroach sql`).

This patch fixes this by also producing the TLS client cert for the
`demo` user.

As a result, running `cockroach sql` against the `demo` cluster
is now possible **without entering a password**.

Passwordless access is also possible from other SQL clients by
providing the `demo` client cert and key in the connection URL.
This is now simplified thanks to change 1 above, as the certificate
directory is now always the same.

**Change 3 (enhancement): show how to use use `cockroach sql`**

Prior to this patch, `demo` would only print out connection URLs on
startup and `\demo ls`. Because the connection URLs are relatively
long, it was making it complex and error-prone to copy-paste them
properly for use in a separate `cockroach sql` session.

This patch prints out an example `cockroach sql` invocation alongside
the connection URLs.

For example:
```
...
(cli) cockroach sql --certs-dir=/data/home/kena/.cockroach-demo -u demo -p 26258
(sql) postgresql://demo:demo99186@127.0.0.1:26258/defaultdb?sslmode=require&sslrootcert=%2Fdata%2Fhome%2Fkena%2F.cockroach-demo%2Fca.crt
...
```

**Change 4 (enhancement): simpler multi-tenant TLS**

Prior to this patch, `cockroach demo` was generating the tenant TLS
configuration using a separate CA key/cert pair. This is not needed
for a demo session, and was also forcing every SQL client that wants
to connect to a secondary tenant server to specify the tenant CA cert
in connection URLs.

This patch simplifies this by reusing the same CA for simulating
secondary tenants. This also enables running `cockroach sql` without
extra arguments to connect to a secondary tenant server.

**Change 5 (enhancement): enable overriding the password using an env var**

As requested in #92541. Although the need for this is now limited
because automated tests can now rely on stable client certs
in `~/.cockroach-demo`.

----

Release note (cli change): The TLS parameters to connect to a
`cockroach demo` cluster from SQL clients has been simplified.

Release note (cli change): The password assigned to the `demo` user in
`cockroach demo` can now be overridden by the env var
`COCKROACH_DEMO_PASSWORD`. This is meant for use in automated tests,
when tests cannot be configured to use TLS client cert authentication
using the client certs in `$HOME/.cockroach-demo`.

### CODE REVIEW

**Comment by knz** [comment]
@aadityasondhi you will also need this. Otherwise the TLS configuration for debug zip will be too complicated.
